lib/intercom/request.rb

```diff
@@ -151,6 +151,9 @@ def raise_application_errors_on_failure(error_list_details, http_code)
       }
       case error_code
       when 'unauthorized', 'forbidden', 'token_not_found'
+        if error_details['message'].include? 'suspended application'
+          raise Intercom::AppSuspendedError.new(error_details['message'], error_context)
+        end
         raise Intercom::AuthenticationError.new(error_details['message'], error_context)
```

Comment:
Can we move raise AuthenticationError into the else block to make it clearer?
```ruby
+        if error_details['message'].include? 'suspended application'
+          raise Intercom::AppSuspendedError.new(error_details['message'], error_context)
          else
           raise Intercom::AuthenticationError.new(error_details['message'], error_context)
+        end
         
```

Reply:
Yup! 👍